Units Move and Attack

kyleslyngstad opened the pull request:
Major Update with basic functionality for moving units and having them
attack.

kyleslyngstad commented:
Started out the trip with no internet, so I coded without seeing your github milestone about using the mouse. Currently the game uses the 'x', 'c' and arrow keys. Can always add functionality for mouse later. I did manage to find an emulator and play the game. I've got a lot better sense of the game now.

alex-nishikawa commented on board_functions.js:
```diff
@@ -1,110 +1,449 @@
+    $(document).keydown(function (event) {
+        event.preventDefault();
+        switch (event.which) {
+            case 37: //Left Arrow
```
If you create a variable for left arrow, (or all the buttons) you won't have to use the comment.

alex-nishikawa commented on board_layouts.js:
```diff
@@ -0,0 +1,117 @@
+//2d array to define what the player has selecting, moving, or attacking
+var move_pos = [];
```
You should try not to pollute the global namespace.  You could store all of these in one global board object:
```
var board = {
    move_pos: [],
    field_info: {
       def: {
             grass: 1
       }
}
```

kyleslyngstad commented:
Added a new commit to create functions for the keys and also reduce the number of variables in the global namespace. Also added Turns and Teams (although right now the player has no way of distinguishing which units are on their team besides clicking on them and seeing if they can move).

alex-nishikawa commented:
:+1: 

Don't merge into master until I say so, I'm trying to setup a deploy pipeline

alex-nishikawa commented:
@kyleslyngstad go ahead and merge since I'm taking too long to setup a deploy method
